eventsapi/models.py

```diff
@@ -1,5 +1,35 @@
-from pydantic import BaseModel
+from typing import Annotated, Literal, Union, Optional
+from pydantic import Field, BaseModel
+from uuid import UUID
 
 
 class DetailMessage(BaseModel):
     detail: str
+
+
+class BaseEvent(BaseModel):
+    course_id: int
+    impression_id: UUID
+    source_uri: str
+    timestamp: int
+
+
+class ContentLoadedV1(BaseEvent):
+    eventname: Literal['content_loaded_v1']
+    content_id: UUID
+    variant: str
+
+
+class ContentLoadFailedV1(BaseEvent):
+    eventname: Literal['content_load_failed_v1']
+    content_id: UUID
+    error: Optional[str] = ''
```

Comment:
What is the difference with using this vs `error: Optional[str]`? Is it just that there will never be a `None` / `null`? If so, would it be better to allow that so we can differentiate when the original event had a string vs not?

Reply:
Ok, we'll modify it. I think based on your previous comment we thought this the intended behavior, but I agree that it would be helpful to know if a comment was actually specified or not in case we want to fix something on the client side.